accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -310,42 +311,51 @@ and $S$ is \VOID{} or \code{\VOID*}.
 ```
 
 _Comparing to Dart before null-safety, this means that it is no longer allowed
-to "return void to null" in a regular function._
+to return a void expression in a regular function if the return type is
+`Null`._
 
-At [this location](https://github.com/dart-lang/language/blob/65b8267be0ebb9b3f0849e2061e6132021a4827d/specification/dartLangSpec.tex#L15525)
+At [this location](https://github.com/dart-lang/language/blob/65b8267be0ebb9b3f0849e2061e6132021a4827d/specification/dartLangSpec.tex#L15507)
 about an asynchronous non-generator function with future value type `$T_v$`,
 the text is changed as follows:
 
 ```
+It is a compile-time error if $s$ is \code{\RETURN{};},
+unless $T_v$
+is \VOID, \DYNAMIC, or \code{Null}.
+%
+It is a compile-time error if $s$ is \code{\RETURN{} $e$;},
+$T_v$ is \VOID,
+and \flatten{S} is neither \VOID, \DYNAMIC, \code{Null},
+\code{\VOID*}, \code{\DYNAMIC*}, nor \code{Null*}.
```

Comment:
This really feels at best pointless to me.  There are hundreds of places in the spec where we talk about `dynamic`, `Null` and `void`, all of which, unless otherwise called out, are intended to apply to `dynamic*`, `Null*` and `void*`.  Likewise for `int` and `int*`.  Shall we go through the spec and point out that every comment about something of type `int` applies equally to something of type `int*`?  I think the intent is clear, and calling it out in some places simply makes things inconsistent and confusing.

Also, we certainly don't intend to put this text into the formal spec as far as I know, since we don't intend to specify the legacy support.

Reply:
Very good, removed the references to `*` types in the text about returns.